exp/internal/controllers/machinepool_controller.go

```diff
@@ -116,6 +116,12 @@ func (r *MachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	if mp.Spec.Replicas == nil {
+		err := errors.Errorf("machinepool %s doesn't have replicas set", mp.Name)
+		log.Error(err, "Failed to find replicas in the MachinePool spec.", mp.Name)
+		return ctrl.Result{}, err
```

Comment:
Let's only return the error here. Controller-runtime logs almost the exact same log message already